Cant add torrents by local file path on windows

## info

160f2917fa85916046c997eccb5858a2e38c592e

OS: Windows

Architecture: X64

## problem


When i try to add a torrent by local file path on windows i get the follow error "failed to fetch torrent"

## backtrace / error messages
"
[NFO] 2023-11-05 21:28:21.0454784 -0300 -03 m=+15.516645601     fetching torrent from c:\Users\re\Downloads\rodrigo.torrent
[ERR] 2023-11-05 21:28:21.0462283 -0300 -03 m=+15.517395501     failed to fetch torrent: Get "c:\\Users\\re\\Downloads\\rodrigo.torrent": unsupported protocol scheme "c"

> try [baa5f67](https://github.com/majestrate/XD/commit/baa5f672a964cc8cce6370bdc796179a90e3ba24)

`lib\util\helpers_windows.go:7:2: package os/path is not in std (D:\devel\go1.21.4.windows-amd64\go\src\os\path)`

--- a/lib/bittorrent/swarm/swarm.go
+++ b/lib/bittorrent/swarm/swarm.go
@@ -374,6 +374,7 @@ func (sw *Swarm) addFileTorrent(path string) (err error) {
 		err = info.BDecode(f)
 		f.Close()
 		if err == nil {
+			log.Infof("fetched torrent from %s, starting allocation", path)
 			var t storage.Torrent
 			t, err = sw.Torrents.st.OpenTorrent(&info)
 			if err == nil {

--- a/lib/bittorrent/swarm/torrent.go
+++ b/lib/bittorrent/swarm/torrent.go
@@ -589,7 +589,7 @@ func (t *Torrent) putInfoSlice(idx uint32, data []byte) {
 				})
 			} else {
 				t.puttingMetaInfo = false
-				log.Errorf("failed to get meta info %s", err.Error())
+				log.Errorf("failed to get meta info: %s", err.Error())
 				t.resetPendingInfo()
 			}
 		} else {

--- a/lib/util/helpers_unix.go
+++ b/lib/util/helpers_unix.go
@@ -1,4 +1,4 @@
-// go:build !windows
+//go:build !windows
 
 package util
 

--- a/lib/util/helpers_windows.go
+++ b/lib/util/helpers_windows.go
@@ -4,7 +4,6 @@ package util
 
 import (
 	"net/url"
-	"os/path"
 	"strings"
 )
 
@@ -14,7 +13,6 @@ func urlSchemePath(u *url.URL, scheme, path *string) {
 		// something like C:/wahtever
 		sch = "file"
 	}
-	*outPath = u.Path
-	*outSceme = sch
-
+	*path = u.String()
+	*scheme = sch
 }

